### ISSUE TITLE
Stats + highscore screen

### ISSUE THREAD

**Issue by Sebeb** [dhubber/mazed-up#8]
Upon either losing or completing the game, the player should be presented with a screen showing their time and the percentage of pills picked up (not counting death pills.) We could either determne a score from these factors, and provide bonuses for lack of damage taken, or simply use the time to determine the players position in the scoreboard. If we go for the latter, it would be nice to have a seperate scoreboard for 100% pill runthroughs, in which the player picks up every non-lethal pill, to keep the hardcore gamers happy.

There are plenty of good, free assets for Unity which allow free hosting of highscores online, although obviously feel free to write your own if you really want. Adding online highscores could definitely boost the game's appeal. It could be fun to tell the player on the main menu exactly where their highest score ranks in the world, to provide extra motivation.

**Comment by dhubber** [comment]
Yes, I agree we need this.  The game has the right difficulty balance that its difficult to complete in the first play-through so some measure of progress, and way of recording it, is important.  You suggested at some point having the death pill from the very first level.  If you want a stat for non-lethal pills, then this is essential otherwise there will be different numbers of non-lethal pills.  I would say hold 3 stats in the high-score table 
1) level reached (highest priority)
2) total no. of pills collected
3) total time to finish game (lowest priority)
So, the progress in going up the high-score table will be Complete game first -> complete game with all pills -> complete game with all pills in the fastest time, which is the correct order we want I believe.

Regarding having scores online, assets/code is one thing but we'd still need some server to put them on.  Is this easy enough to do, and more importantly, free??

**Comment by Sebeb** [comment]
Definitely agree on the priorities.
In the latest version, the death pill always spawns in the first level of the game, and the player is required to pick it up to proceed. Meaning that while they do lose one life at the beginning, they will definitely know what the death pill looks like for the rest of the game. Regarding the stats, I was planning on just not counting the death pills in calculations. So if in 10 levels, 35 pills are spawned, 5 of which are death pills, and the player picks up 25 pills, including 1 death pill in the first level, the total percent of pills picked up will be calculated from 24 of 30. I'm not sure whether depicting this information as a percentage or as actual numbers is the best way to go.
I've heard http://dreamlo.com is very straight forward to use for highscore hosting, and it's compeltely free!

**Comment by Sebeb** [comment]
I actually changed this, so the death pill does count exactly the same as any other pill. But its spawn is no longer as random as other pills. It now definitely spawns in the first and last levels, and then at random levels in between. But the total amount of death pill spawns can only ever be one less than the player's starting lives, meaning that while it is possible to take every pill, the player can't take any damage other than death pill damage when doing so. Should make getting that 100% score all the more challenging :grin:
